### PR TITLE
 Add checks for pointsAwarded

### PR DESCRIPTION
**Pull request by nigoshh**
If the student has already completed the exercise, a green check sign
is always displayed in the top right corner. When loading the page,
a completed exercise renders the complete screen with "practice more"
("harjoittele lisää" in Finnish) but without the progress bar.
When the student completes an exercise, the complete screen includes
the progress bar.

### CODE REVIEW

File: src/components/music/Chord.js [hunk 1]
```diff
@@ -54,7 +54,13 @@ export default class Chord {
    * @returns ["root"]
    */
   getCorrectAnswerKeys(answer, correctAnswer) {
-    if (!answer.root || !correctAnswer) return []
+    if (
```
Comment (AuroraLS3) [inline src/components/music/Chord.js:57]: Good fix

File: src/components/music/Interval.js [hunk 2]
```diff
@@ -61,7 +61,13 @@ export default class Interval {
    * @returns ["interval"]
    */
   getCorrectAnswerKeys(answer, correctAnswer) {
-    if (!answer.interval || !correctAnswer) return []
+    if (
```
Comment (AuroraLS3) [inline src/components/music/Interval.js:64]: Good fix

File: src/components/music/Interval.js [hunk 3]
```diff
@@ -88,16 +94,14 @@ export default class Interval {
 
   makeAnswerPayload(answer, correctAnswer, correct) {
     const answerQualityLabel = qualities[answer.quality].label.toLowerCase()
-    const answerIntervalLabel = intervalLabels[
-      answer.interval
-    ].label.toLowerCase()
+    const answerIntervalLabel = intervalLabels[answer.interval].toLowerCase()
```
Comment (AuroraLS3) [inline src/components/music/Interval.js:97]: Good fix

File: src/components/music/MusicExerciseWrapper.js [hunk 4]
```diff
@@ -124,20 +132,38 @@ class MusicExerciseWrapper extends React.Component {
...
 
-  onCorrectAnswer = payload => {
-    this.setState({ correctAnswers: this.state.correctAnswers + 1 })
+  onCorrectAnswer = async payload => {
+    const correctAnswers = this.state.correctAnswers + 1
+
+    if (!this.context.loggedIn) {
+      this.setState({
+        correctAnswers,
+        completed: correctAnswers >= this.state.requiredAnswers,
```
Comment (AuroraLS3) [inline src/components/music/MusicExerciseWrapper.js:144]: Nice fix!

File: src/components/music/MusicExerciseWrapper.js [hunk 5]
```diff
@@ -124,20 +132,38 @@ class MusicExerciseWrapper extends React.Component {
...
+      })
+      return
+    }
+
+    // update correctAnswers already here (instead of after waiting for the
+    // response from backend) to show progress in the progress bar more quickly
```
Comment (AuroraLS3) [inline src/components/music/MusicExerciseWrapper.js:150]: Nice!

File: src/components/music/Scale.js [hunk 6]
```diff
@@ -60,7 +60,13 @@ export default class Scale {
    * @returns ["root"]
    */
   getCorrectAnswerKeys(answer, correctAnswer) {
-    if (!answer.root || !correctAnswer) return []
+    if (
```
Comment (AuroraLS3) [inline src/components/music/Scale.js:63]: Great fix!

**Review by AuroraLS3 — APPROVED** [review]
Nice work managing the ton of bugs you had to fix!

**Review by rescawen — APPROVED** [review]
Finally after many days of hard work!

**Pull request merged**